### ISSUE TITLE
Catenate cell values when creating scalar variable in variable table

### ISSUE THREAD

**Issue by pekkaklarck** [robotframework/robotframework#1914]
Currently this this invalid:

``` robotframework
*** Variables ***
${SCALAR}    values    in    multiple    cells
...    possibly   on    multiple    rows
```

That used to create `${SCALAR}` variable with a list value, but that support was removed in RF 2.8 (#939) after being deprecated already in RF 2.5 (#401). If a list is needed, a list variable can be created instead.

Now in RF 2.9 we can change the above to work so that individual cells are catenated together. That will ease creating scalar variables with long string values. Catenating with space as a separator would probably work best in most cases, but making it possible to specify a custom separator by having `SEPARATOR=<sep>` in the first cell, exactly like with `Catenate` keyword, would allow using custom separators like newline or empty string.

See also related issue #1919 about `Set Test/Suite/Global Variable` keywords.

**Comment by pekkaklarck** [comment]
We originally planned to change `Set Test/Suite/Global Variable` keywords to work this way in RF 2.9. Unfortunately they still allow creating scalar lists like this, and thus the change would have been quite big. Instead we decided just to remove that old and useless functionality (#1919).

**Comment by pekkaklarck** [comment]
This functionality has already been implemented (with tests) in 41b63613fb85fe1e67cd09ea075fb57516a726c2. Still needs to be documented in the User Guide.
